### DISCUSSION TITLE
Add osu!catch poison fruit/mines

### DISCUSSION THREAD

**Discussion by chrsbell** [ppy/osu#13036] [Ideas]
**Describe the new feature:** 

I think we could spice up osu!catch mapping and the gamemode in general by adding poisonous fruits, or mines, which will break your combo if you touch them. They could also possibly have some other effect like causing a random AR change for a certain amount of seconds to throw off players. Mappers could use the poison fruits to add visual flair to their maps as well as restrict movement without changing the CS.

**Proposal designs of the feature:** 

A simple design of a regular fruit but with poison oozing from it. Of course this should be skinnable. Note that the poison fruits would need to be mapped specifically for ctb, so they wouldn't exist in converts. That pretty much covers what I had in mind, please share ideas and criticism!

**Comment by 59xa** [comment]
In my opinion, this should be a _mod_ instead of implementing it as a part of the original gameplay.

**Reply by iSimpFor** [reply]
i'm not in favor of implementing as only a mod because i don't think a mod like this could ever place/use mines like a mapper could

i would agree with this as an extra mod with regular implementation

**Reply by 59xa** [reply]
Don't forget about the already existing beatmaps that were made before this is implemented, so having this as a regular ruleset implementation won't make any sense. [As mentioned here](https://github.com/ppy/osu/discussions/13036#discussioncomment-2263147), it should be instead randomly generated by the game rather than the mapper having to manually place them; hence the reason why a mod implementation is more understandable than altering the already-existing rulesets.

**Comment by iSimpFor** [comment]
i think the mines should be extended to all rulesets as notes a mapper can place down, hitting the mines like a regular note will trigger the mine

in standard, a mine can be triggered by tapping the note but it can be decided to trigger just hovering over it 

mines will only appear as a single regular note and not a slider

if a mappers don't like mines, they don't have to add them but they can add them as a challenge to dodge such notes if they desire

i disagree with what mines should do as you said, i think mines shouldn't break your combo but you will lose hp if you hit them and there should be a stat for how many mines you avoid across the whole map, there can be an extra effect such as a random AR change for a short amount of time or something else as you said

a mine can count for combo but that's up for the devs to decide

there should also be a mod to remove mines from a map that comes with a score penalty

**Reply by 59xa** [reply]
**I do approve of your approach on this proposal, however I have another iteration of how this could work instead:**

> if a mappers don't like mines, they don't have to add them but they can add them as a challenge to dodge such notes if they desire

If a mapper does not want to include mines on their beatmaps, then it shouldn't really be a manual implementation. Having the game place poison fruit/mines throughout the beatmap would be a preferrable choice. Hence as mentioned from my response above leans more toward having it as a mod implementation.

> i disagree with what mines should do as you said, i think mines shouldn't break your combo but you will lose hp if you hit them and there should be a stat for how many mines you avoid across the whole map, there can be an extra effect such as a random AR change for a short amount of time or something else as you said

I agree with the player losing health the moment they tap/hover the mine, however, a sudden AR change as a random effect should not exist if it were to be a regular ruleset implementation. Keep in mind as well that previously-made beatmaps will not have this implementation if it were to be added afterwards, so keep it the same as it should be.

> there should also be a mod to remove mines from a map that comes with a score penalty

If you think about it, it should be a complete mod implementation instead of altering/changing existing rulesets in my opinion.

**Reply by iSimpFor** [reply]
> If a mapper does not want to include mines on their beatmaps, then it shouldn't really be a manual implementation. Having the game place poison fruit/mines throughout the beatmap would be a preferrable choice. Hence as mentioned from my response above leans more toward having it as a mod implementation.

i don't see anything wrong with regular ruleset implementation, a person can just not use the mines if they didn't want them in their map since a freshly created map will start with no notes placed and people who want to use the mines will use them, it will not be required for mines to be used either
i believe a mapper who wants to use mines will generally create more interesting patterns of mines than what a game algorithm can do, the reason i suggested a extra mine mod to go with the ruleset implantation is to give players a option to play maps without mines with mines although the mines will not be as well placed as a mapper who actually knows how to use the mines(it's possible that this mod is unnecessary  in case of ruleset implementation though)

> I agree with the player losing health the moment they tap/hover the mine, however, a sudden AR change as a random effect should not exist if it were to be a regular ruleset implementation. Keep in mind as well that previously-made beatmaps will not have this implementation if it were to be added afterwards, so keep it the same as it should be.

the AR change is just an example of a effect, there can be a different effect that doesn't impact the previously-made beatmaps negatively 

> If you think about it, it should be a complete mod implementation instead of altering/changing existing rulesets in my opinion.

see above and elaborate on your side

**Reply by 59xa** [reply]
> i don't see anything wrong with regular ruleset implementation, a person can just not use the mines if they didn't want them in their map since a freshly created map will start with no notes placed and people who want to use the mines will use them, it will not be required for mines to be used either

Having it implemented as a regular implementation will cause a huge difference for upcoming maps with the altered ruleset and the already-existing ones that do not have it. If it were to be implemented as a mod, every beatmap (upcoming and existing) could have it in sacrifice of a well-designed beatmap level made to challenge the user. Implementing this as a ruleset change will make the older beatmaps feel like they're out-of-date.

> i believe a mapper who wants to use mines will generally create more interesting patterns of mines than what a game algorithm can do, the reason i suggested a extra mine mod to go with the ruleset implantation is to give players a option to play maps without mines with mines although the mines will not be as well placed as a mapper who actually knows how to use the mines(it's possible that this mod is unnecessary in case of ruleset implementation though)

One thing I want to mention for this is that the mapping community for each ruleset should be consulted first if they are in favour of having this proposal implemented. Also consult the players if they want to add this feature as it could have implications on the difficulty of the beatmap. We don't want the existing playerbase to be frustrated and outraged over the sudden ruleset change.

> the AR change is just an example of a effect, there can be a different effect that doesn't impact the previously-made beatmaps negatively

As mentioned above, consult both mappers and the playerbase. There is a **percentage of players who play this game casually**, we do not want to suddenly increase the difficulty of upcoming beatmaps. Doing this as a ruleset implementation will be disliked by some of the playerbase instead unless if it were to be an implemented mod instead.

**Reply by iSimpFor** [reply]
i think it would be best if the userbase was polled 
we can consider a poll for each ruleset and userbase(players and mappers) (this will probably be 8 polls)
